scripts/google/covid_mobility/covidmobility.py

```diff
@@ -0,0 +1,233 @@
+    sub_region_2: str = row[3] # Whitespaces are okay here
+    dcid: str = get_dcid_from_region(sub_region_2, sub_region_1, country_code) # country/ESP is an example
+
+    # Return a dictionary of the row
+    return {
+        'dcid': dcid,
+        'country_code': country_code,
+        'country': country,
+        'sub_region_1': sub_region_1,
+        'sub_region_2': sub_region_2,
+        'date': row[4],
+        'LocalBusiness': row[5],
+        'GroceryStore&Pharmacy': row[6],
+        'Park': row[7],
+        'TransportHub': row[8],
+        'Workplace': row[9],
+        'Residence': row[10]
+    }
+
+def generate_places_node_ids_for_row(row_data: dict) -> dict:
+    """Given row data, return the node local ids for each place
+
+    Args:
+        row_data (dict): row_data in the form of an object
+
+    Returns:
+        node_ids: object containing place -> node Id
+    """
+    location_name: str = row_data['sub_region_2'] + row_data['sub_region_1'] + row_data['country']
+    # Ignore any non-ascii characters
+    location_name: str = ''.join([i if ord(i) < 128 else '_' for i in location_name])
+    # Get rid of any whitespaces
+    location_name: str = location_name.replace(" ", "")
+    location_dcid: str = row_data['dcid']
+
+    return {
+        "LocalBusiness": f"{location_name}LocalBusiness",
+        "GroceryStore&Pharmacy": f"{location_name}GroceryStore&Pharmacy",
+        "Park": f"{location_name}Park",
+        "TransportHub": f"{location_name}TransportHub",
+        "Workplace": f"{location_name}Workplace",
+        "Residence": f"{location_name}Residence"
+    }
+
+def get_dcid_from_region(sub_region_2: str, sub_region_1: str, country_code: str) -> str:
+    """
+    Convert region to dcid.
+    :param sub_region_2: Usually a US County (only US County allowed for now)
+    :param sub_region_1: State or Province (only US State allowed for now)
+    :param country_code: Country Code as Alpha2
+    :return location_dcid: the dcid of the location
+    :except KeyError if any of the regions aren't hashable to dcid
+    """
+
+    # Counties
+    if sub_region_2:
+        if country_code != 'US':
+            raise KeyError('sub_region_2 is only supported for US Counties.')
+        state_alpha_code: str = name_to_alpha2.USSTATE_MAP[sub_region_1]
+        dcid: str = county_to_dcid.COUNTY_MAP[state_alpha_code][sub_region_2]
+    # States or Provinces
+    elif sub_region_1:
+        if country_code != 'US':
+            raise KeyError('sub_region_1 is only supported for US States.')
+        region_alpha_code: str = name_to_alpha2.USSTATE_MAP[sub_region_1]
+        dcid: str = alpha2_to_dcid.USSTATE_MAP[region_alpha_code]
+    # Countries
+    else:
+        dcid: str = alpha2_to_dcid.COUNTRY_MAP[country_code]
+    return dcid
+
+def write_statistical_pop_node(mobility_locations: dict, output_path: str) -> None:
+    """
+    Generates unique ID's for each location and writes a Statistical Population Node in MCF format.
```

Comment:
Instead of having two functions, you can also do string templating for the node ID

e.g. in the Population template, instead of `Node: {local_id}`

`Node: {geoId}_{category_name}`

Reply:
Similar for Obs

Reply:
done